python-checks/src/main/java/org/sonar/python/checks/tests/DedicatedAssertionCheck.java

```diff
@@ -0,0 +1,239 @@
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.checks.tests;
+
+import java.util.List;
+import java.util.Map;
+import javax.annotation.CheckForNull;
+import org.sonar.check.Rule;
+import org.sonar.plugins.python.api.PythonSubscriptionCheck;
+import org.sonar.plugins.python.api.SubscriptionContext;
+import org.sonar.plugins.python.api.tree.Argument;
+import org.sonar.plugins.python.api.tree.BinaryExpression;
+import org.sonar.plugins.python.api.tree.CallExpression;
+import org.sonar.plugins.python.api.tree.Expression;
+import org.sonar.plugins.python.api.tree.IsExpression;
+import org.sonar.plugins.python.api.tree.Name;
+import org.sonar.plugins.python.api.tree.QualifiedExpression;
+import org.sonar.plugins.python.api.tree.RegularArgument;
+import org.sonar.plugins.python.api.tree.Tree;
+import org.sonar.python.checks.Expressions;
+import org.sonar.python.tests.UnittestUtils;
+import org.sonar.python.tree.TreeUtils;
+
+
+@Rule(key = "S5906")
+public class DedicatedAssertionCheck extends PythonSubscriptionCheck {
+
+  private static final String MESSAGE = "Consider using \"%s\" instead.";
+  private static final String MESSAGE_ROUND_ALMOST_EQUAL = "Consider using the \"places\" argument of \"%s\" instead.";
+
+  private static final String ASSERT_TRUE = "assertTrue";
+  private static final String ASSERT_FALSE = "assertFalse";
+  private static final String ASSERT_EQUAL = "assertEqual";
+  private static final String ASSERT_NOT_EQUAL = "assertNotEqual";
+  private static final String ASSERT_ALMOST_EQUAL = "assertAlmostEqual";
+  private static final String ASSERT_NOT_ALMOST_EQUAL = "assertNotAlmostEqual";
+
+  private static final String IS = "#is#";
+  private static final String IS_NOT = "#is-not#";
+  private static final String IN = "#in#";
+
+  private static final String FIRST = "first";
+  private static final String SECOND = "second";
+
+  private static final String ROUND = "round";
+  private static final String IS_INSTANCE = "isinstance";
+
+  private static final Map<String, String> ASSERT_TRUE_SUGGESTIONS = Map.of(
+    "==", ASSERT_EQUAL,
+    "!=", ASSERT_NOT_EQUAL,
+    ">", "assertGreater",
+    ">=", "assertGreaterEqual",
+    "<", "assertLess",
+    "<=", "assertLessEqual",
+    IN, "assertIn",
```

Comment:
Nitpick, sound a bit strange to have sometimes constant defined values and sometimes a String literal.
I believe it is because the constant defined values are used later in the switch ?
Not sure if my remark is relevant, just about uniformity.

Reply:
That's a good question, indeed for operator it feel more readable to keep them as it is.
Also I'm afraid that if we replace everything by constant, the class may start with 20+ lines of such definitions, which may not be desirable.
We can either let as it is, replace only method name or replace everything, I let you see what is the more readable.

Reply:
Then I think I'll let it as it is. It's not ideal but no the solution is, really 😄.